features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/OtelTracerBuilderProviderTest.kt

```diff
@@ -0,0 +1,363 @@
+            InternalLogger.Target.USER,
+            OtelTracerProvider.TRACING_NOT_ENABLED_ERROR_MESSAGE
+        )
+    }
+
+    @Test
+    fun `M log a developer error W build { default service name not available }`() {
+        // GIVEN
+        whenever(mockSdkCore.service) doReturn ""
+
+        // WHEN
+        testedOtelTracerProviderBuilder.build()
+
+        // THEN
+        mockInternalLogger.verifyLog(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.USER,
+            OtelTracerProvider.DEFAULT_SERVICE_NAME_IS_MISSING_ERROR_MESSAGE
+        )
+    }
+
+    // endregion
+
+    // region ID generation
+
+    @Test
+    fun `M build tracers which generate Spans with 64 bits long ids W build`(
+        @StringForgery(type = StringForgeryType.ALPHA_NUMERICAL) spanName: String
+    ) {
+        // Given
+        val tracerProvider = testedOtelTracerProviderBuilder
+            .build()
+
+        // When
+        val span = tracerProvider.tracerBuilder(fakeInstrumentationName).build().spanBuilder(spanName).startSpan()
+
+        // Then
+        assertThat(span.spanContext.spanIdBytes.size).isEqualTo(8)
+    }
+
+    @Test
+    fun `M build tracers which generate Spans with 128 bits long trace ids W build`(
+        @StringForgery(type = StringForgeryType.ALPHA_NUMERICAL) spanName: String
+    ) {
+        // Given
+        val tracerProvider = testedOtelTracerProviderBuilder.build()
+
+        // When
+        val span = tracerProvider.tracerBuilder(fakeInstrumentationName).build().spanBuilder(spanName).startSpan()
+        val traceId = span.spanContext.traceId
+        // Then
+        assertThat(traceId.substring(0, 16)).isEqualTo("0000000000000000")
+        assertThat(span.spanContext.traceIdBytes.size).isEqualTo(16)
+    }
+
+    // endregion
+
+    // region Tracer creation
+    @Test
+    fun `M build spans using the instrumentationName W build`(
+        @StringForgery(type = StringForgeryType.ALPHA_NUMERICAL) spanName: String
+    ) {
+        // Given
+        val tracerProvider = testedOtelTracerProviderBuilder.build()
+
+        // When
+        val span = tracerProvider.tracerBuilder(fakeInstrumentationName).build().spanBuilder(spanName).startSpan()
+        val traceId = span.spanContext.traceId
+        // Then
+        assertThat(traceId.substring(0, 16)).isEqualTo("0000000000000000")
+        assertThat(span.spanContext.traceIdBytes.size).isEqualTo(16)
+    }
+
+    @Test
+    fun `M create a single tracer for same instrumentation name W get`() {
+        // Given
+        val tracerProvider = testedOtelTracerProviderBuilder.build()
+        val tracer = tracerProvider.get(fakeInstrumentationName)
+
+        // Then
+        repeat(20) {
+            val newTracer = tracerProvider.get(fakeInstrumentationName)
+            assertThat(newTracer).isSameAs(tracer)
+        }
+    }
+
+    @Test
+    fun `M create a single tracer for same instrumentation name W get { same instrumentationVersion }`(
```

Comment:
should we add the test when instrumentation version is different? by looking on the source code, if we already have a tracer for the given instrumentation and we provide different instrumentation version, it will be ignored (which is a bit confusing, so maybe we should add a log entry?).

Reply:
we could yes...